gpt4all-api/gpt4all_api/Dockerfile.buildkit

```diff
@@ -0,0 +1,20 @@
+# syntax=docker/dockerfile:1.0.0-experimental
+FROM tiangolo/uvicorn-gunicorn:python3.11
+
+# Put first so anytime this file changes other cached layers are invalidated.
+COPY gpt4all_api/requirements.txt /requirements.txt
+
+RUN pip install --upgrade pip
+
+# Run various pip install commands with ssh keys from host machine.
+RUN --mount=type=ssh pip install -r /requirements.txt && \
+  rm -Rf /root/.cache && rm -Rf /tmp/pip-install*
+
+# Finally, copy app and client.
+COPY gpt4all_api/app /app
+
+RUN mkdir -p /models
+
+# Include the following line to bake a model into the image and not have to download it on API start.
+RUN wget https://gpt4all.io/models/ggml-gpt4all-j-v1.3-groovy.bin -P /models
```

Comment:
Would it be possible to make this model value (ggml-gpt4all-j-v1.3-groovy.bin) a defaulted build arg? Thus, allowing devs to optionally specify a different default model when building the docker image?

```
ARG MODEL_BIN=ggml-gpt4all-j-v1.3-groovy.bin
# Include the following line to bake a model into the image and not have to download it on API start.
RUN wget https://gpt4all.io/models/$MODEL_BIN -P /models
```

Then, optionally, the build command could be
```
DOCKER_BUILDKIT=1 docker build -t gpt4all_api --progress plain -f gpt4all_api/Dockerfile.buildkit --build-arg MODEL_BIN=ggml-gpt4all-l13b-snoozy.bin .

//or, just use the default

DOCKER_BUILDKIT=1 docker build -t gpt4all_api --progress plain -f gpt4all_api/Dockerfile.buildkit  .
```

Reply:
will do

Reply:
I also had this mod locally, to pass in the ARG, 
but also trim the wget output which is too verbose (substituted with the MD5 to confirm everything ok (md5 is in the model db json)
```
ARG MODEL_BIN=ggml-gpt4all-j-v1.3-groovy.bin
RUN wget -q --show-progress=off https://gpt4all.io/models/${MODEL_BIN} -P /models \
  && md5sum /models/${MODEL_BIN}
```

Reply:
addressed